### PR TITLE
test: add broker integration tests

### PR DESCRIPTION
**Pull request by bayological**
### Description
Add integration tests tying all McMint related work together.

Things to test:
- [x] Stable to Collateral swaps
- [x] Stable to Stable swaps
- [x] Bucket updates
- [x] Discovery functions 

### Other changes

N/A

### Tested

It is tests

### Related issues

- Fixes https://github.com/mento-protocol/mento-general/issues/43

### Backwards compatibility

N/A

### Documentation

N/A

### CODE REVIEW

File: test/integration/BrokerIntegration.t.sol [hunk 1]
```diff
@@ -0,0 +1,292 @@
...
+      bucketIn = exchange.bucket0;
+      bucketOut = exchange.bucket1;
+    } else {
+      bucketIn = exchange.bucket1;
+      bucketOut = exchange.bucket0;
+    }
+
+    // Get expected out.. ¯\_(ツ)_/¯
+    // Maybe use broker.getAmountOut
+    expectedOut = exchange.pricingModule.getAmountOut(bucketIn, bucketOut, exchange.config.spread.unwrap(), amountIn);
```
Comment (bowd) [inline test/integration/BrokerIntegration.t.sol:67]: Yeah I think using `broker.getAmountOut` here is fair.
I don't think we should make expectations on the amounts themselves in the test, that's part of the components. 
But in an integration test I think it's fair to test that `broker.getAmountOut` ends up being the `amountOut` of the immediate swap.

**Review by bowd — APPROVED** [review]
LGTM, I would indeed use `broker.getAmountOut` just to make it feel more consistent as an integration test, testing the boundary of the system.

But yeah a big bunch of the setup there is my own code so I don't think it makes sense for me to review it, so I'm hoping somebody else put some eyes on it.

**Review by philbow61 — COMMENTED** [review]
Looks good to me👌🏻💪🏻
